### PR TITLE
[Kotlin migration] AnkiPackageExporterTest

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description
Kotlin migration of AnkiPackageExporterTest
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt [hunk 1]
```diff
@@ -13,175 +13,163 @@
...
-        List<String> files = Arrays.asList(new File(unzipDirectory).list());
-        File fileNames[] = new File[3];
-        int i = 0;
-        for (String x : files) {
-            File f = new File(unzipDirectory + "/" + x);
-            fileNames[i++] = f;
+        val files = Arrays.asList(File(unzipDirectory).list())
+        val fileNames = arrayOfNulls<File>(3)
+        var i = 0
+        for (x in files) {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt:100]: add a cleanup to use a mapIndexed

File: AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt [hunk 2]
```diff
@@ -13,175 +13,163 @@
...
-
-        assertEquals("", res);
-    }
+    fun stripHTML_will_remove_html_with_unicode_whitespace() {
+        val exporter: Exporter = exporterForDeckWithMedia
+        val text = """
+
+ 	<[sound:test.mp3]>
+
	 <!-- Comment
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt:122]: Nope, I'm sorry, but this is not readable. I believe automated translation correctly transformed unicode code in its unicode equivalent. But there is no easy way for a human reading code to see all symbols here. 
There is even a risk that some automated cleaning tool want to remove the tab, and that it's not catched at review.

Actually, I'd even love if you'd consider adding a cleanup to document what are the various unicode characters we use here.

File: AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt [hunk 3]
```diff
@@ -13,175 +13,163 @@
...
-                "\n\u2029" +
-                "\t\u2004" +
-                "<!-- Comment \n \u1680 --> <\"tag\\n><style><s>";
-        final String res = exporter.stripHTML(text);
-
-        assertEquals("", res);
-    }
+    fun stripHTML_will_remove_html_with_unicode_whitespace() {
+        val exporter: Exporter = exporterForDeckWithMedia
+        val text = """
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt:119]: cleanup `@language("html")`

File: AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt [hunk 4]
```diff
@@ -13,175 +13,163 @@
...
 
+        assertEquals("", res)
+    }
 
-    private void checkMediaExportStringIs(File[] files, String s) throws IOException {
-        for (File f : files) {
-            if (!"media".equals(f.getName())) {
-                continue;
+    @Throws(IOException::class)
+    private fun checkMediaExportStringIs(files: Array<File?>, s: String) {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt:130]: Can you have `Array<File>` instead?

**Review by Arthur-Milchior — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt (hunk 4 above)
Comment (Prince-kushwaha) [reply AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt:133]: not possible to make array not nullable

**Comment by Prince-kushwaha** [comment]
@david-allison Doesn't compile is due to CI https://github.com/ankidroid/Anki-Android/issues/11091#issuecomment-1130226317

**Comment by mikehardy** [comment]
@Prince-kushwaha 

> Doesn't compile is due to CI [#11091 (comment)](https://github.com/ankidroid/Anki-Android/issues/11091#issuecomment-1130226317)


Nope - that's not how it works

1- it is compiling (now at least?), so "doesn't compile" is not the correct statement. "Doesn't pass unit tests..." is the correct statement

2- your linked comment is a flaky test I am tracking, but you are failing on **exactly the code you altered** not that test.

Something you have done here has broken unit tests somehow

```

AnkiPackageExporterTest > fileInExportIsCopied FAILED
    java.lang.AssertionError: 
    Expected: iterable with items ["collection.anki2", "media", "0"] in any order
         but: not matched: ["collection.anki2", "0", "media"]
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at com.ichi2.libanki.AnkiPackageExporterTest.fileInExportIsCopied(AnkiPackageExporterTest.kt:109)
AnkiPackageExporterTest > missingFileInDeckExportDoesSkipsFile FAILED
    java.lang.AssertionError: 
    Expected: iterable with items ["collection.anki2", "media"] in any order
         but: not matched: ["collection.anki2", "media"]
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at com.ichi2.libanki.AnkiPackageExporterTest.missingFileInDeckExportDoesSkipsFile(AnkiPackageExporterTest.kt:76)
```

**Comment by david-allison** [comment]
Superseded by https://github.com/ankidroid/Anki-Android/pull/12100

**Pull request closed without merging**